driver-core/src/main/com/mongodb/operation/FindAndDeleteOperation.java

```diff
@@ -169,28 +180,69 @@ public BsonDocument getSort() {
+     * @return this
+     * @since 3.2
+     * @mongodb.server.release 3.2
+     */
+    public FindAndDeleteOperation<T> writeConcern(final WriteConcern writeConcern) {
+        this.writeConcern = writeConcern;
+        return this;
+    }
+
     @Override
     public T execute(final WriteBinding binding) {
-        return executeWrappedCommandProtocol(binding, namespace.getDatabaseName(), getFindAndRemoveDocument(),
-                                             CommandResultDocumentCodec.create(decoder, "value"),
-                                             FindAndModifyHelper.<T>transformer());
+        return withConnection(binding, new CallableWithConnection<T>() {
+            @Override
+            public T call(final Connection connection) {
+                return executeWrappedCommandProtocol(binding, namespace.getDatabaseName(), asCommandDocument(connection.getDescription()),
+                        CommandResultDocumentCodec.create(decoder, "value"),
+                        FindAndModifyHelper.<T>transformer());
+            }
+        });
     }
 
     @Override
     public void executeAsync(final AsyncWriteBinding binding, final SingleResultCallback<T> callback) {
-        executeWrappedCommandProtocolAsync(binding, namespace.getDatabaseName(), getFindAndRemoveDocument(),
-                                           CommandResultDocumentCodec.create(decoder, "value"),
-                                           FindAndModifyHelper.<T>transformer(), callback);
+        withConnection(binding, new AsyncCallableWithConnection() {
+                    @Override
+                    public void call(final AsyncConnection connection, final Throwable t) {
+                        if (t != null) {
+                            errorHandlingCallback(callback).onResult(null, t);
+                        } else {
+                            executeWrappedCommandProtocolAsync(binding, namespace.getDatabaseName(),
+                                    asCommandDocument(connection.getDescription()),  CommandResultDocumentCodec.create(decoder, "value"),
+                                    FindAndModifyHelper.<T>transformer(),
+                                    releasingCallback(errorHandlingCallback(callback), connection));
+                        }
+                    }
+                });
     }
 
-    private BsonDocument getFindAndRemoveDocument() {
+    private BsonDocument asCommandDocument(final ConnectionDescription description) {
         BsonDocument command = new BsonDocument("findandmodify", new BsonString(namespace.getCollectionName()));
         putIfNotNull(command, "query", getFilter());
         putIfNotNull(command, "fields", getProjection());
         putIfNotNull(command, "sort", getSort());
         putIfNotZero(command, "maxTimeMS", getMaxTime(MILLISECONDS));
         command.put("remove", BsonBoolean.TRUE);
+        if (writeConcern != null & serverIsAtLeastVersionThreeDotTwo(description)) {
```

Comment:
This should be:

     serverIsAtLeastVersionThreeDotTwo(description) && !writeConcern.isServerDefault() && writeConcern.isAcknowledged()

as I don't think the driver should be sending 

    writeConcern : { w : 0}

to the server.